resolwe/storage/models.py

```diff
@@ -243,6 +246,44 @@ def transfer_data(self, destination: "StorageLocation"):
             ReferencedPath.objects.bulk_create(referenced_paths)
             destination.files.add(*referenced_paths)
 
+    def verify_data(self) -> bool:
+        """Verify data stored in this location.
+
+        Verify hashes and sizes of all files. This operation could be slow
+        and cause network traffic, use with care.
+
+        :returns: True if data is OK, False otherwise. In case of failure
+        additional information about error is logged.
+        """
+        hash_types = ["md5", "crc32c", "awss3etag"]
+        return_value = True
+        for referenced_path in self.files:
+            url = self.get_path(filename=referenced_path.path)
+            hashes = dict(zip(hash_types, self.connector.get_hashes(url, hash_types)))
+            for hash_name, hash_value in hashes:
+                referenced_path_hash = getattr(referenced_path, hash_name)
+                if not referenced_path_hash:
+                    logger.warning(
+                        "ReferencedPath with id {} has no {} hash".format(
+                            referenced_path.id, hash_name
+                        )
+                    )
+                    continue
+
+                # Hashes differ: log error and continue to check other hashes.
+                if referenced_path != hash_value:
+                    return_value = False
+                    logger.error(
+                        "ReferencedPath with id {} has wrong {} hash: {} instead of {}".format(
+                            referenced_path.id,
```

Comment:
The string has 4 placeholders and 5 values to insert.

Reply:
Fixed.